Добавлен контроллер и вид для отображения ошибок

toyrik opened the pull request:
В файл web/index.php добавлен перехват всплывающих ощибок и передача их
в ErrorHandler для обработки
В файле web/autoload.php добавлен перехват ошибки подключения файла

toyrik committed:
Добавлен контроллер и вид для отображения ошибок

В файл web/index.php добавлен перехват всплывающих ощибок и передача их
в ErrorHandler для обработки
В файле web/autoload.php добавлен перехват ошибки подключения файла

toyrik committed:
Добавляем возможность создания ползовательских обработчиков исключений.

В конфиге приложения добавлен раздел для обявления классов обработчиков.
Добавлен пример пользовательского обработчика ошибок.

vedro-compota commented on web/autoload.php:
```diff
@@ -1,10 +1,12 @@
 <?php
```
откатить правки, тут ничего не надо

vedro-compota commented on application/views/error.php:
```diff
@@ -0,0 +1,21 @@
+
+<?php
+use ItForFree\SimpleMVC\DebugPrinter;
```
предлагаю откатить отдачу заголовка вообще

vedro-compota reviewed (changes requested):
требуются правки, добавлены комментарии к разным фрагментам

toyrik committed:
Убираем зависимости от классов ядра в autoload.php

Убираем использование ErrorHanler из ядра в файле autoload.php
В файле views/error.php убрана отдача заголовка.

toyrik committed:
Исправляем досадную опечатку в файле конфигурации

vedro-compota commented on web/autoload.php:
```diff
@@ -20,8 +20,7 @@ function autoload($className)
     //echo $fileName;
-  
-    require $fileName;
+    require_once $fileName;
```
или пояснение оптимизации или просьба отменить изменения

vedro-compota commented on web/index.php:
```diff
@@ -1,13 +1,21 @@
 <?php
 
```
откатить все в этом файле

vedro-compota reviewed (changes requested):
просьба ещё поправить

toyrik committed:
Откатываем правки в autoload.php и index.php

toyrik committed:
Откатываем правки в autoload.php и index.php

toyrik committed:
Merge branch 'fix403exeption' of github.com:toyrik/SimpleMVC-example into fix403exeption

toyrik commented:
Поправил

vedro-compota commented on web/autoload.php:
```diff
@@ -20,7 +20,7 @@ function autoload($className)
     $fileName .= str_replace('_', DIRECTORY_SEPARATOR, $className) . '.php';
      
     //echo $fileName;
-  
```
в этом файле полностью отменяем изменения

vedro-compota commented on web/index.php:
```diff
@@ -8,6 +8,7 @@
     $localConfig);
 
 
```
аналогично откатываем изменения

toyrik committed:
Откатываем ненужные изменения

vedro-compota reviewed (changes requested):
в папке docs/  в корне просьба создать файл ExeptionsHandling.md и описать там функционал, счанала преамбула, а потом пояснения с цитатами кода с указанием путей к файлам.

toyrik committed:
Добавляем документацию

toyrik commented:
То есть что то [подобное](https://github.com/it-for-free/SimpleMVC-example/pull/6/commits/51800f760d00911eea93bb8f5330a36663d0adda), я правильно понимаю?

toyrik committed:
Добавляем описание метода run()

toyrik committed:
Добавляем описание создание собственного обработчика
